### ISSUE TITLE
.tscn cannot save floats in exponential format

### ISSUE THREAD

**Issue by bojidar-bg** [godotengine/godot#3527]
Steps to reproduce:

1. New scene
2. Add a Node2D root, and a sprite + an animation player
3. Make a new (autoplay) animation, and animate sprite:transform/scale
4. Set sprite's scale to `(0,0)`, it will be changed to some healthy minimum, and add keyframe
5. Move the animation player to another time, and add keyframe for `(1,1)`
6. Play the animation in the editor -> everything works just fine (sprite starts "invisible", finishes normal)
7. Save as `.xscn` and play that scene only -> everything works just fine
8. Save as `.tscn` and play that scene only -> sprite starts big, finishes normal
9. Edit first key frame manually to become `(0,0)` and not `(0.00001,0.00001)` (this is the workaround ATM)
10. Resave `.tscn` and play that scene only -> everything works just fine

I think the bug comes from the fact that `0.00001` is saved as `1e-05`, but read as `105` in game, because:

* When printing the scale every `_fixed_process`, I got:
````
# .xscn (step 7)
0.00001,0.00001
0.00001,0.00001
#...
1,1
# .tscn (step 8)
105,105
105,105

#...
1,1
````
* Comparing the corresponding lines between the files:
````
# .tscn (step 8)
tracks/1/interp = 1
tracks/1/keys = { "cont":true, "times":FloatArray( 0, 0.6 ), "transitions":FloatArray( 1, 1 ), "values":[ Vector2( 1e-05, 1e-05 ), Vector2( 1, 1 ) ] }

# .xscn (step 7)
                <int name="tracks/1/interp"> 1 </int>
                <dictionary name="tracks/1/keys" shared="false">
                        <string> "cont" </string>
                        <bool> True </bool>
                        <string> "times" </string>
                        <real_array  len="2">                           0, 0.6 </real_array>
                        <string> "transitions" </string>
                        <real_array  len="2">                           1, 1 </real_array>
                        <string> "values" </string>
                        <array  len="2" shared="false">
                                <vector2> 1e-05, 1e-05 </vector2>
                                <vector2> 1, 1 </vector2>
                        </array>
                </dictionary>
````

Also, are is a few screenshots of the bug:
* While the animation is running: (it also animates opacity, so it is semi-transparent)
![image](https://cloud.githubusercontent.com/assets/5276727/12702179/19f5e15e-c829-11e5-81de-de12c41b77b4.png)
* After the animation ran: (look at this circle)
![image](https://cloud.githubusercontent.com/assets/5276727/12702182/31e0581c-c829-11e5-96fa-0af2ff6704cd.png)

**Comment by bojidar-bg** [comment]
Additional steps:
1. Return keyframe 1 to `(0.00001,0.00001)`
2. Resave `.tscn`, and reload scene / close it and open it again / restart godot
3. See than animation is corrupted now and play it in editor -> sprite starts big, finishes normal

**Comment by akien-mga** [comment]
Fixed by 9386b76, reduz got the reference wrong ;)
